xray/commands/audit/sca/java/mvn.go

```diff
@@ -1,98 +1,146 @@
+}
+
+func (mdt *MavenDepTreeManager) execMavenDepTree(depTreeExecDir string) ([]byte, error) {
+	mavenDepTreePath := filepath.Join(depTreeExecDir, mavenDepTreeOutputFile)
+	goals := []string{"com.jfrog:maven-dep-tree:" + mavenDepTreeVersion + ":" + mdt.cmdName, "-DdepsTreeOutputFile=" + mavenDepTreePath}
+	if err := mdt.runMvnCmd(goals); err != nil {
+		return nil, err
 	}
+
+	mavenDepTreeOutput, err := os.ReadFile(mavenDepTreePath)
+	err = errorutils.CheckError(err)
+	return mavenDepTreeOutput, err
 }
 
-func runMvn(buildConfiguration *utils.BuildConfiguration, insecureTls, ignoreConfigFile, useWrapper bool, mvnProps map[string]any) (err error) {
-	goals := []string{"-B", "compile", "test-compile", "-Dcheckstyle.skip", "-Denforcer.skip"}
-	log.Debug(fmt.Sprintf("mvn command goals: %v", goals))
-	configFilePath := ""
-	if !ignoreConfigFile {
-		var exists bool
-		configFilePath, exists, err = utils.GetProjectConfFilePath(utils.Maven)
-		if err != nil {
-			return
-		}
-		if exists {
-			log.Debug("Using resolver config from " + configFilePath)
-		}
+func (mdt *MavenDepTreeManager) runMvnCmd(goals []string) error {
+	if mdt.settingsXmlPath != "" {
+		goals = append(goals, "-s", mdt.settingsXmlPath)
 	}
-	if useWrapper {
-		useWrapper, err = isMavenWrapperExist()
-		if err != nil {
-			return
-		}
-		if mvnProps == nil {
-			mvnProps = make(map[string]any)
+
+	//#nosec G204
+	cmdOutput, err := exec.Command("mvn", goals...).CombinedOutput()
+	if err != nil {
+		if len(cmdOutput) > 0 {
+			log.Info(string(cmdOutput))
 		}
-		mvnProps["useWrapper"] = useWrapper
+		err = fmt.Errorf("failed running command 'mvn %s': %s", strings.Join(goals, " "), err.Error())
 	}
-	// Read config
-	vConfig, err := utils.ReadMavenConfig(configFilePath, mvnProps)
+	return err
+}
+
+// Creates a new settings.xml file configured with the provided server and repository from the current MavenDepTreeManager instance.
+// The settings.xml will be written to the given path.
+func (mdt *MavenDepTreeManager) createSettingsXmlWithConfiguredArtifactory(path string) error {
+	username, password, err := mdt.server.GetAuthenticationCredentials()
 	if err != nil {
 		return err
 	}
-	mvnParams := mvnutils.NewMvnUtils().
-		SetConfig(vConfig).
-		SetBuildConf(buildConfiguration).
-		SetGoals(goals).
-		SetInsecureTls(insecureTls).
-		SetDisableDeploy(true)
-	return mvnutils.RunMvn(mvnParams)
-}
-
-func isMavenWrapperExist() (bool, error) {
-	wrapperName := "mvnw"
-	if coreutils.IsWindows() {
-		wrapperName += ".cmd"
+	remoteRepositoryFullPath, err := url.JoinPath(mdt.server.ArtifactoryUrl, mdt.depsRepo)
+	if err != nil {
+		return err
 	}
-	return fileutils.IsFileExists(wrapperName, false)
+	mdt.settingsXmlPath = filepath.Join(path, settingsXmlFile)
+	settingsXmlContent := fmt.Sprintf(settingsXmlTemplate, username, password, remoteRepositoryFullPath)
+	return errorutils.CheckError(os.WriteFile(mdt.settingsXmlPath, []byte(settingsXmlContent), 0666))
```

Comment:
This file contains secrets
```suggestion
	return errorutils.CheckError(os.WriteFile(mdt.settingsXmlPath, []byte(settingsXmlContent), 0600))
```